joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -701,7 +702,7 @@ controller_interface::CallbackReturn JointTrajectoryController::on_configure(
   joints_angle_wraparound_.resize(dof_);
   for (size_t i = 0; i < dof_; ++i)
   {
-    if (!urdf_.empty())
+    if (!get_robot_description().empty())
```

Comment:
shouldn't we check here even for `model_` being empty? Can it be that the variable changes between `on_init` and `on_configure`? When a new robot_description is received, maybe?  but the `urdf::Model` doesn't have an empty check, we would need a persistent variable like `urdf_initialized`.

Reply:
The new `robot_description`, will be only transmitted to the new controller only upon unloading and loading the controller again, so I don't think this will change once the controller is initialized. Moreover, checking the model is valid or not can be good and be a part of a different PR.

Reply:
Ok, fine for me to fix this in a different PR. I just started thinking about that reading it now (I know that I've written the current code ;)). Does it make sense to parse this in on_init at all? Currently, we use the `model_` variable only in the on_configure method.

Reply:
Yes! As we have the URDF in the init method, we could initialize the model in the init method and reuse it